# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/gc/Marking.cpp
# Commit: b29a261a06ec
# Full Hash: b29a261a06ec664f02c57121ce6e5302590fa7ac
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2022-03-04 04:18:28
# Regressor Bug: 1643586
# File Overlap Count: 1
# Description:
#   Bug 1643586 - Part 4: Remove flag for disabling compartment checks from JSContext r=jandem
#   
#   This is only used in one place and it's simpler to check the tracer kind there.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D140191
# ==============================================================================

diff -r 619f2aabe4d5 -r b29a261a06ec js/src/gc/Marking.cpp
--- a/js/src/gc/Marking.cpp	Thu Mar 03 17:49:17 2022 +0000
+++ b/js/src/gc/Marking.cpp	Thu Mar 03 17:49:17 2022 +0000
@@ -554,11 +554,15 @@
     MOZ_ASSERT_IF(gcMarker->tracingZone,
                   (*dst)->zone() == gcMarker->tracingZone);
   }
+
+  // Skip compartment checks for this edge.
+  if (trc->kind() == JS::TracerKind::CompartmentCheck) {
+    return;
+  }
 #endif
 
   // Clear expected compartment for cross-compartment edge.
   AutoClearTracingSource acts(trc);
-  AutoDisableCompartmentCheckTracer adcct;
   TraceEdgeInternal(trc, ConvertToBase(dst->unbarrieredAddress()), name);
 }
 template void js::TraceSameZoneCrossCompartmentEdge(